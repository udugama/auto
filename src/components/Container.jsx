import React from 'react';
import axios from 'axios';
import { Paper, Grid } from '@material-ui/core';

import SearchBox from './SearchBox';
import Navigate from './Navigate';
import Content from './Content';

export default class Container extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
          currentPage: 1,
          search: '',
          numberOfPages: 0,
          hitsPerPage: 20,
        };

        this.requestTimer = '';

    }

    searchStories (currentPageNumber) { 
        const url = `https://hn.algolia.com/api/v1/search?query=${this.state.search}&tags=story&page=${currentPageNumber}`;
        axios.get(url)
        .then(response => {
            const {
                status,
                data: {
                    nbPages,
                    hits,
                }
            } = response;
            if (status === 200) {
                let totalPages = nbPages;
                if (totalPages > 1) {
                    totalPages = totalPages - 1;
                }
                this.setState({
                    numberOfPages: totalPages,
                    data: hits,
                });
            }
        });
    }

    delayedRequest (event) {
        this.setState({ search: event.target.value });
        if (this.requestTimer) {
            clearTimeout(this.requestTimer);
        }
        this.requestTimer = setTimeout(() => {
            this.setState({ currentPage: 1 });
            this.searchStories(1);
        }, 200);
    }

    handleNextPage() {
        const totalPages = this.state.numberOfPages;
        if (this.state.currentPage < totalPages) {
            const currentPageNumber = this.state.currentPage + 1;
            this.setState({ currentPage: currentPageNumber });
            this.searchStories(currentPageNumber);
        }
    }

    handlePreviousPage() {
        if (this.state.currentPage > 1) {
            const currentPageNumber = this.state.currentPage - 1;
            this.setState({ currentPage: currentPageNumber });
            this.searchStories(currentPageNumber);
        }
    }

    render() {
        return ( 
            <Paper >
                <Grid container spacing={3}>
                <Grid item xs={12}>
                    <Paper >
                        <SearchBox handleOnChange={this.delayedRequest.bind(this)}  />
                    </Paper>
                </Grid>
                <Grid item xs={1}></Grid>
                <Grid item xs={10}>
                    <Content content={this.state.data} />
                </Grid>
                <Grid item xs={1}></Grid>
                <Grid item xs={1} >
                    <Paper ></Paper>
                </Grid>
                <Grid item xs={3} >
                    <Paper ><Navigate name={'Previous'} onClick={this.handlePreviousPage.bind(this)} /></Paper>
                </Grid>
        
                <Grid item xs={3} >
                    <Paper > Page {this.state.currentPage} of {this.state.numberOfPages} </Paper>
                </Grid>
                <Grid item xs={3} >
                    <Paper ><Navigate name={'Next'} onClick={this.handleNextPage.bind(this)} /></Paper>
                </Grid>
                <Grid item xs={1} >
                    <Paper ></Paper>
                </Grid>
                </Grid>
    
            </Paper>
        );
    }
}