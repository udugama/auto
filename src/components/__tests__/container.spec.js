import React from 'react';
import ReactDOM from 'react-dom';
import Container from './../Container';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';


it('renders without crashing', () => {
    shallow(<Container />);
});

it('render Container initialy', () => {
  const { getByText } = render(<Container />);
  expect(getByText).toMatchSnapshot();
});