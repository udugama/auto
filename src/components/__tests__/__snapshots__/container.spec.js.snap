// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Container initialy 1`] = `[Function]`;
